app/services/users/omniauth.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Users::Omniauth
+  attr_reader :auth
+
+  def initialize(auth)
+    @auth = auth
+  end
+
+  def get
+    user
+  end
+
+  private
+
+  delegate :provider, :uid, :info, to: :auth
+  delegate :email, to: :info
+
+  def user
+    user = User.where(provider: provider, uid: uid, email: email).first
+    user ||= User.new(
+      provider: provider,
+      uid: uid,
+      email: email,
+      password: Devise.friendly_token[0, 20]
+    )
+    user.skip_confirmation!
+    user.save
+    user
```

Comment:
@sinpai currently `#user` method is overloaded with logic, let's split it to several methods